src/redux/slices/stillageSlice.ts

```diff
@@ -0,0 +1,32 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { IShelfsResponse } from '../../api/intefaces'
+
+export interface StillageState {
+  list: IShelfsResponse[] | null
+}
+
+const initialState: StillageState = {
+  list: null,
+}
+
+export const stillageSlice = createSlice({
+  name: 'stillage',
+  initialState,
+  reducers: {
+    setStillageList: (state, action: { payload: IShelfsResponse[]; type: string }) => {
+      state.list = action.payload
+    },
+    clearStillageList: (state) => {
+      state.list = null
+    },
+    removeStillageItem: (state, action: { payload: string; type: string }) => {
+      if (state.list) {
+        state.list = state.list.filter((shelf) => shelf.id !== action.payload)
+      }
```

Comment:
In this case, I would use [Nullish coalescing assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment) or [Logical AND assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND_assignment):

```javascript
state.list ??= state.list.filter((shelf) => shelf.id !== action.payload)
```

Reply:
This change will make the code less readable for my team, so I will leave it as it is.